File: Networking/src/main/java/server/RpcConcurrentServer.java
package server;

import rpc.ClientRpcWorker;
import services.IServices;

import java.net.Socket;

public class RpcConcurrentServer extends AbsConcurrentServer {
    private final IServices server;

    public RpcConcurrentServer(int port, IServices server) {
        super(port);
        this.server = server;
        System.out.println("RpcConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        ClientRpcWorker worker = new ClientRpcWorker(server, client);

        return new Thread(worker);
    }
}

File: Networking/server/RpcConcurrentServer.cs
﻿using System;
using System.Net.Sockets;
using System.Threading;
using Networking.rpc;
using Services;

namespace Networking.server
{
    public class RpcConcurrentServer : AbsConcurrentServer
    {
        private readonly IServices _server;

        public RpcConcurrentServer(string host, int port, IServices server) : base(host, port)
        {
            _server = server;
            Console.WriteLine("RpcConcurrentServer...");
        }

        protected override Thread CreateWorker(TcpClient client)
        {
            ClientRpcWorker worker = new(_server, client);
            return new Thread(new ThreadStart(worker.Run));
        }
    }
}
